src/compression/index.js

```diff
@@ -1,29 +1,24 @@
-import RawDecoder from './raw';
-import LZWDecoder from './lzw';
-import JpegDecoder from './jpeg';
-import DeflateDecoder from './deflate';
-import PackbitsDecoder from './packbits';
-import LercDecoder from './lerc';
+const registry = new Map();
 
-export function getDecoder(fileDirectory) {
-  switch (fileDirectory.Compression) {
-    case undefined:
-    case 1: // no compression
-      return new RawDecoder();
-    case 5: // LZW
-      return new LZWDecoder();
-    case 6: // JPEG
-      throw new Error('old style JPEG compression is not supported.');
-    case 7: // JPEG
-      return new JpegDecoder(fileDirectory);
-    case 8: // Deflate as recognized by Adobe
-    case 32946: // Deflate GDAL default
-      return new DeflateDecoder();
-    case 32773: // packbits
-      return new PackbitsDecoder();
-    case 34887: // LERC
-      return new LercDecoder(fileDirectory);
-    default:
-      throw new Error(`Unknown compression method identifier: ${fileDirectory.Compression}`);
+export function addDecoder(cases, importFn) {
```

Comment:
So the idea is that the built in compression algorithms are provided by default, but users can add their own?

How does this work with bundling? I guess there will be some split files involved, right?

Reply:
> So the idea is that the built in compression algorithms are provided by default, but users can add their own?

Exactly. For example, if a user overrides `lzw` with a different module at runtime (`addDecoder`), the final bundle will have two separate chunks (the default, and the override) but only the latter gets loaded at runtime.

> How does this work with bundling? I guess there will be some split files involved, right?

It depends on the bundler, and is configurable in many (as mentioned above).